### PR TITLE
Document WebResponseObject/BasicHtmlWebResponseObject properties

### PR DESCRIPTION
**Pull request by kevinoid**
# PR Summary

Improve xmldoc comments for properties of `WebResponseObject` and `BasicHtmlWebResponseObject`.

## PR Context

I found the current documentation to be unhelpful for understanding the values returned by `Invoke-WebRequest`.  This PR contains my current understanding after examining the code.  Feel free to modify or rewrite to better fit your preferred style.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [x] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

**Comment by adityapatwardhan** [comment]
@PoshChan please rebuild all

**Comment by PoshChan** [comment]
@adityapatwardhan, successfully started rebuild of `PowerShell-CI-static-analysis, PowerShell-CI-Windows, PowerShell-CI-macOS, PowerShell-CI-Linux`

**Comment by kevinoid** [comment]
I should also mention that `.Content` and `.RawContentStream` contain only the response content, while `.RawContent` contains status, headers, and content.  I tried to make that clear in the comments, but I suspect it will still catch users by surprise.  Assuming this is intentional and none of the properties will be deprecated, would *bold* be appropriate to draw attention to this difference?

**Comment by kevinoid** [comment]
Also, since `BasicHtmlWebResponseObject` doesn't override `.ToString()`, it will differ from `.Content` and `.RawContent` since it contains the response content decoded as ASCII with non-printable characters removed.  Although it meets the definition of a "string representation", which is all that it is documented to return, I worry it may also catch users by surprise if they write code which uses `ToString()`.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/BasicHtmlWebResponseObject.Common.cs [hunk 1]
```diff
@@ -99,7 +105,7 @@ public WebCmdletElementCollection InputFields
         private WebCmdletElementCollection _links;
 
         /// <summary>
-        /// Gets the Links property.
+        /// Gets the HTML a elements parsed from <see cref="Content"/>.
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/BasicHtmlWebResponseObject.Common.cs]:
```suggestion
        /// Gets the HTML a links parsed from <see cref="Content"/>.
```

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/BasicHtmlWebResponseObject.Common.cs [hunk 2]
```diff
@@ -57,22 +57,28 @@ public BasicHtmlWebResponseObject(HttpResponseMessage response, Stream contentSt
...
+        /// Encoding of the response body from the <c>Content-Type</c> header,
+        /// or <c>null</c> if the encoding could not be determined.
         /// </value>
         public Encoding Encoding { get; private set; }
 
         private WebCmdletElementCollection _inputFields;
 
         /// <summary>
-        /// Gets the Fields property.
+        /// Gets the HTML input elements parsed from <see cref="Content"/>.
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/BasicHtmlWebResponseObject.Common.cs]:
```suggestion
        /// Gets the HTML input fields parsed from <see cref="Content"/>.
```

**Review by adityapatwardhan — APPROVED** [review]

**Comment by adityapatwardhan** [comment]
@kevinoid Thank you for your contribution!

**Comment by kevinoid** [comment]
My pleasure.  Thanks for reviewing and merging @adityapatwardhan!

**Pull request merged**